chore: internal docs for beta (#187)
storyblok/storyblok-cli: src/README.md

![Storyblok ImagoType](https://raw.githubusercontent.com/storyblok/.github/refs/heads/main/profile/public/github-banner.png)

> # Storyblok CLI Package Reference

Reviewer: ```suggestion
# Storyblok CLI
```

## Installation

> For the latest beta version, install the package using the following command:
> 
> ```bash
> npm install storyblok@beta
> ```
> 
> Or for an specific beta version:
> 
> ```bash
> npm install storyblok@4.0.0-beta.<version>
> ```

Reviewer: My suggestion is to keep the non-beta installation command in here. It makes sense for people to want the stable version for a new project.

```suggestion
Add the package to your project with the following command:

```bash
npm install storyblok@latest
```

You can also install the beta version, currently in development:

```bash
npm install storyblok@beta
```

It's also possible to target a specific beta version:

```bash
npm install storyblok@4.0.0-beta.<version>
```
```

Reviewer 2: Hey @alvarosabu, is this readme going to be in `main`, or in a beta branch? If it's supposed to be in `main`, I agree with @jeremenichelli.

Author: No, it's going to live on the `next` branch @manuelschroederdev @jeremenichelli these documentations are strictly for the beta since they will be replaced by official documentation later on

Reviewer 2: That's what I thought. 😄 Then feel free to disregard the suggestion from Jeremias here.

> ## Command Reference

Reviewer: Probably `## API` is a better title for this section.

| [`login`](./commands/login/README.md) | ✅ Ready | Improved DX and credentials storage in ~/.storyblok/credentials.json |
| [`logout`](./commands/logout/README.md) | ✅ Ready | |
| [`user`](./commands/user/README.md) | ✅ Ready | |
| [`languages pull`](./commands/languages/README.md) | ✅ Ready | Replaces previous pull-languages |
| [`components pull`](./commands/components/pull/README.md) | ✅ Ready | Replaces previous pull-components |
| [`components push`](./commands/components/push/README.md) | ✅ Ready | Replaces previous push-components. Also handles dependencies such as groups, tags, presets and whitelists. (Datasources is pending) |
| `components delete` | 📝 Planned | Will replace delete-component and delete-components |
| [`migrations generate`](./commands/migrations/generate/README.md) | ✅ Ready | Replaces previous generate-migrations |
| [`migrations run`](./commands/migrations/run/README.md) | ✅ Ready | Replaces previous run-migrations |
| [`migrations rollback`](./commands/migrations/rollback/README.md) | ✅ Ready | Replaces previous rollback-migrations |
| [`types generate`](./commands/types/generate/README.md) | ✅ Ready | Replaces previous generate-typescript-typedefs |

> | `sync` | ⚠️ v3 | |

Reviewer: I suggest leaving only the emojis, and adding a legend in the end or before the table.

Reviewer 2: @jeremenichelli I'm also not a fan of emojis, but I think they serve a purpose here.

Author: My intention with the emojis was to help the user to quickly identify the status visually when reading through the table. Normally it would be done with CSS but we don't have that flavor on markdown.

Considering this is temporal documentation, I wouldn't be strict about this part.

| `datasources pull` | 📝 Planned | |
| `datasources push` | 📝 Planned | |
| `datasources delete` | 📝 Planned | Will replace delete-datasources |
| `select` | 💬 TBD | To be discussed |
| `quickstart` | 💬 TBD | To be discussed |
| `spaces` | 💬 TBD | To be discussed |
| `import` | 💬 TBD | To be discussed |

## Global Options

> These options can be used with any command:

Reviewer: ```suggestion
These options are available for all commands:
```

| Option | Description | Default |
|--------|-------------|---------|
| `-v, --verbose` | Enable verbose output for debugging | `false` |
| `--ci` | Enable CI mode (coming soon) | `false` |

> [!TIP]

> > When reporting a bug or opening a support ticket, please run the command with the `--verbose` flag to provide detailed information that will help us better understand and resolve the issue.

Reviewer: ```suggestion
> When reporting a bug or opening a support ticket, please run the command with the `--verbose` flag and add the output to it. This will help us better understand and resolve the issue.
```
